src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -1896,10 +1899,12 @@ FixedwingPositionControl::tecs_update_pitch_throttle(float alt_sp, float airspee
 	}
 
 	/* tell TECS to update its state, but let it know when it cannot actually control the plane */
-	bool in_air_alt_control = (!_vehicle_land_detected.landed &&
-				   (_control_mode.flag_control_auto_enabled ||
-				    _control_mode.flag_control_velocity_enabled ||
-				    _control_mode.flag_control_altitude_enabled));
+	bool in_air_alt_control = ((!_vehicle_land_detected.landed ||
+				    (_launch_detection_state == LAUNCHDETECTION_RES_DETECTED_ENABLEMOTORS
+				     && mode == tecs_status_s::TECS_MODE_TAKEOFF))
```

Comment:
This is a mouthful, should we have a helper function? Same logic could be used for run_tecs above?

Reply:
e.g. just an extra bool calculated directly above?
```
bool is_launching = _launch_detection_state == LAUNCHDETECTION_RES_DETECTED_ENABLEMOTORS && mode == tecs_status_s::TECS_MODE_TAKEOFF;
bool in_air_alt_control = !_vehicle_land_detected.landed || is_launching;
```

or do you mean an encapsulated function call?
```
bool in_air_alt_control = !_vehicle_land_detected.landed || is_launching();
```

Reply:
The first one is good, I'd say.